Add docs for raster datasets (#849)
koordinates/kart: docs/pages/raster_datasets.rst

Working with Raster Datasets
====================================

.. Note:: Before using this quick guide, it will help your understanding to do the :doc:`Basic Usage Tutorial </pages/basic_usage_tutorial>`.

Dataset type
~~~~~~~~~~~~

> Raster datasets are currently implemented as :doc:`Raster Datasets V1 </pages/development/raster_v1>`. Kart version controls raster tiles themselves as GeoTIFF files, which means the tiles are checked out into the working copy without any conversion step - the contents of the working copy is an accurate reflection of the contents of the dataset itself at a particular point in time.

Reviewer: ```suggestion
Kart internally stores raster tiles as GeoTIFF files, which means the tiles are checked out into the working copy without any conversion step - the contents of the working copy is an accurate reflection of the contents of the dataset itself at a particular point in time.
 
To find out more about the internal storage of raster datasets, see :doc:`Raster Datasets V1 </pages/development/raster_v1>`.
```

I mostly want to de-emphasise the linked doc for beginning users - it's probably not relevant to them / might be confusing

Importing raster datasets
~~~~~~~~~~~~~~~~~~~~~~~~~

- ``kart import <tile> [<tile>] [<tile>] [--convert-to-cog/--preserve-format]``
- ``kart import --dataset=<dataset_name> <tile> [<tile>] [<tile>] [--convert-to-cog/--preserve-format]``

> This command imports one or more raster tiles into the Kart repository in the
> working directory. All tiles are imported into the same dataset.

Reviewer: ```suggestion
This command imports one or more raster tiles into the Kart repository. All tiles are imported into the same dataset.
```

The tiles to import must be GeoTIFF files of some variant - and cloud-optized GeoTIFFs (COGs) are allowed.

If ``--convert-to-cog`` is specified, all tiles that are not cloud-optimized will be converted to cloud-optimized as they are imported, and the dataset will
also have a constraint attached which means that only cloud-optimized tiles can be added. This constraint can be dropped later if needed.

If ``--preserve-format`` is specified, all tiles will be imported as-is, without any conversion step.

If neither option is specified, Kart will prompt to see which you prefer.

Importing tiles as cloud-optimized is recommended, since web-viewers of GeoTIFFs only work on tiles that have been cloud-optimized. If you ever decide to publish your repository or some part of it on the internet, it will be easier at that time if the entire revision history of that repository is all in a single, web-viewable format.

On the other hand, converting non-COG tiles to the COG format will make the import take longer and use more disk space.

For more information, see :ref:`Import vectors / tables into an existing repository`.

Working copy
~~~~~~~~~~~~

> Raster datasets are stored in a file-system based working copy associated with the Kart repository. This working copy is found in folders inside the Kart repository, one folder for each raster dataset, with the same name as the dataset. Since the file system working copy is simply in the Kart repository, it doesn't need to be configured, and you don't need any special software to connect to it.

Reviewer: ```suggestion
Raster datasets are stored in a filesystem-based working copy associated with the Kart repository. This working copy is found in folders inside the repository, one folder for each raster dataset, with the same name as the dataset.
```

Not sure that last sentence adds anything
